sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs

```diff
@@ -102,31 +100,6 @@ internal static void ThrowIfCancellationRequestedOrTimeout(CancellationToken ori
         }
     }
 
-#pragma warning disable SA1402 // File may only contain a single type
-    internal class AzureCorePipelineExecutor : PipelineEnumerator
-#pragma warning restore SA1402 // File may only contain a single type
-    {
-        private readonly HttpMessage _message;
-        private ReadOnlyMemory<HttpPipelinePolicy> _policies;
-
-        public AzureCorePipelineExecutor(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> policies)
-        {
-            _policies = policies;
-            _message = message;
-        }
-        public override bool ProcessNext()
-        {
-            _policies.Span[0].Process(_message, _policies.Slice(1));
-            return true;
-        }
-
-        public async override ValueTask<bool> ProcessNextAsync()
-        {
-            await _policies.Span[0].ProcessAsync(_message, _policies.Slice(1)).ConfigureAwait(false);
-            return true;
-        }
-    }
-
 #pragma warning disable SA1402 // File may only contain a single type
     internal class AzureCoreResponseBufferingPolicy : ResponseBufferingPolicy
```

Comment:
I am still not sure this is the right extensibility model, but it may hinge on @KrzysztofCwalina's ideas about freezing the pipeline, so I'd like to understand these better before moving to far forward with this.